cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/CruiseControlApiImpl.java

```diff
@@ -176,15 +175,22 @@ public CruiseControlApiImpl(Vertx vertx) {
                         String userTaskID = response.getHeader(CC_REST_API_USER_ID_HEADER);
                         response.bodyHandler(buffer -> {
                             JsonObject jsonUserTask = buffer.toJsonObject().getJsonArray("userTasks").getJsonObject(0);
-                            JsonObject json = new JsonObject()
-                                    .put("Status", jsonUserTask.getString("Status"))
-                                    .put("summary", ((JsonObject) Json.decodeValue(jsonUserTask.getString("originalResponse"))).getJsonObject("summary"));
-                            if (json.containsKey(CC_REST_API_ERROR_KEY)) {
-                                result.fail(json.getString(CC_REST_API_ERROR_KEY));
-                            } else {
-                                CruiseControlUserTaskResponse ccResponse = new CruiseControlUserTaskResponse(userTaskID, json);
-                                result.complete(ccResponse);
+                            // This should not be an error with a 200 status but we play it safe
+                            if (jsonUserTask.containsKey(CC_REST_API_ERROR_KEY)) {
+                                result.fail(jsonUserTask.getString(CC_REST_API_ERROR_KEY));
```

Comment:
Same comment, I think we need a `CruiseControlException` or similar. Also is this error key the only piece of relevant info to include?

Reply:
I will add the request path to the exception message to aid debugging.